### PR TITLE
make jms.listener component independent from gateway component

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/service/RevokedTokenDataImpl.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.wso2.carbon.apimgt.gateway.jwt.RevokedJWTDataHolder;
+import org.wso2.carbon.apimgt.impl.token.RevokedTokenService;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+
+public class RevokedTokenDataImpl implements RevokedTokenService {
+
+    public void addRevokedJWTIntoMap(String revokedToken, Long expiryTime) {
+        RevokedJWTDataHolder.getInstance().addRevokedJWTToMap(revokedToken, expiryTime);
+        // Add revoked token to
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/service/RevokedTokenDataImpl.java:13]: comment not completed.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/token/RevokedTokenService.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.wso2.carbon.apimgt.impl.token;
+
+public interface RevokedTokenService {
+
+    public void addRevokedJWTIntoMap(String revokedToken, Long expiryTime);
+
+    public void revokedTokenFromGatewayCache(String accessToken, boolean isJwtToken);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/token/RevokedTokenService.java:7]:
```suggestion
    public void removeTokenFromGatewayCache(String accessToken, boolean isJwtToken);
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/throttling/APIThrottleDataService.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/*
+ *
+ *   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/throttling/APIThrottleDataService.java:3]: license year wrong

File: components/apimgt/org.wso2.carbon.apimgt.jms.listener/src/main/java/org/wso2/carbon/apimgt/jms/listener/ThrottlingRunTimeException.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/*
+*  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.jms.listener/src/main/java/org/wso2/carbon/apimgt/jms/listener/ThrottlingRunTimeException.java:2]: license year

**Pull request merged**